src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -574,6 +575,51 @@ public static void ValidateRequirement(this DicomDataset dataset, DicomTag tag,
         }
     }
 
+    /// <summary>
+    /// Returns a dicom item if it matches a large object size criteria.
+    /// </summary>
+    /// <param name="dataset"></param>
+    /// <param name="minLargeObjectsizeInBytes"></param>
+    /// <param name="maxLargeObjectsizeInBytes"></param>
+    /// <param name="largeDicomItem"></param>
+    /// <returns>dicom item</returns>
+    public static bool TryGetLargeDicomItem(
```

Comment:
I think I had a related question, but I cannot find it now. Do we have code to ensure that we're not dropping any private tags from the metadata when we extract the pixel data?

Reply:
We are not extracting the pixel data, we are only finding the first 4MB or the first set of dicom item as part of initial block. I have unit tests to make sure the block lengths is equal to the original stream length